Assets/Scripts/PlayerController/PlayerHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerHandler : MonoBehaviour, IInteractor
+{
+    [SerializeField]
+    private PlayerMovement m_PlayerMovement;
+
+    [SerializeField]
+    private PlayerAnimation m_PlayerAnimation;
+
+    [SerializeField]
+    private PowerupActor m_PowerupActor;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        m_PlayerAnimation = GetComponent<PlayerAnimation>();
+        m_PlayerMovement = GetComponent<PlayerMovement>();
+        m_PowerupActor = GetComponent<PowerupActor>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        UpdateVelocity();
+    }
+
+    public void HandleInteraction(IInteractable interactable) {
+        if (interactable is JumpPowerUp) {
+            m_PowerupActor.JumpHigher(((JumpPowerUp) interactable).GetBuffDuration());
+        } 
+
+        if (interactable is SpeedPowerUp) {
+            m_PowerupActor.GoFaster(((SpeedPowerUp) interactable).GetBuffDuration());
+        }
+    }
```

Comment:
Like mentioned before, this class should handle the interactions themselves, so they should actually just check for trigger/distance etc and then call the interact method of the interactable associated to it.

Reply:
Ok swee understooded